mtp/types.go

```diff
@@ -116,7 +116,8 @@ type StorageInfo struct {
 }
 
 func (d *StorageInfo) IsHierarchical() bool {
-	return d.FilesystemType == FST_GenericHierarchical
+	return (d.FilesystemType == FST_GenericHierarchical ||
+		d.FilesystemType == FST_DCF)
```

Comment:
since this is in the MTP library, better add a IsDCF method, and do the || in the caller.